x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -710,6 +710,49 @@ class PackagePolicyService {
     }
   }
 
+  public async enrichPolicyWithDefaultsFromPackage(
+    soClient: SavedObjectsClientContract,
+    newPolicy: NewPackagePolicy
+  ): Promise<NewPackagePolicy> {
+    let newPackagePolicy: NewPackagePolicy = newPolicy;
+    if (newPolicy.package) {
+      const newPP = await this.buildPackagePolicyFromPackage(soClient, newPolicy.package.name);
+      if (newPP) {
+        if (newPolicy.package.version !== newPP.package?.version) {
+          throw new Error(
```

Comment:
I added this check in case the `package.version` from the request body is different than the installed version. 
The installed version is queried in [buildPackagePolicyFromPackage](https://github.com/elastic/kibana/blob/dbf47c685a9374535c9c9d2fd97157c58d2fd076/x-pack/plugins/fleet/server/services/package_policy.ts#L717) which I'm using to enrich the defaults.
Is there any use case to allow creating a `package_policy` with different version than the installed base package?